http/http.go

```diff
@@ -57,28 +73,83 @@ func (s Handler) Handle(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	host, port, _ := net.SplitHostPort(req.RemoteAddr)
-	s.Log = s.Log.WithValues("host", host, "port", port)
-	m := path.Dir(req.URL.Path)
-	if strings.HasPrefix(m, "/") {
-		m = trimFirstRune(path.Dir(req.URL.Path))
-	}
+	log := s.Log.WithValues("host", host, "port", port)
 	// If a mac address is provided, log it. Mac address is optional.
-	// Mac address would be useful for logging and tracing.
-	mac, _ := net.ParseMAC(m)
-	s.Log = s.Log.WithValues("mac", mac)
+	mac := parseMac(req.URL.Path)
+	log = log.WithValues("mac", mac.String())
+	filename := filepath.Base(req.URL.Path)
+
+	// clients can send traceparent over HTTP by appending the traceparent string
+	// to the end of the filename they really want
+	longfile := filename // hang onto this to report in traces
+	ctx, shortfile, err := extractTraceparentFromFilename(context.Background(), filename)
+	if err != nil {
+		log.Error(err, "failed to extract traceparent from filename")
+	}
+	if shortfile != filename {
+		log.Info("traceparent found in filename", "filename_with_traceparent", longfile, "filename", shortfile)
+		filename = shortfile
+	}
 
-	got := filepath.Base(req.URL.Path)
-	file, found := binary.Files[got]
+	tracer := otel.Tracer("HTTP")
+	_, span := tracer.Start(ctx, "HTTP get",
+		trace.WithSpanKind(trace.SpanKindServer),
+		trace.WithAttributes(attribute.String("filename", filename)),
+		trace.WithAttributes(attribute.String("requested-filename", longfile)),
+		trace.WithAttributes(attribute.String("ip", host)),
+		trace.WithAttributes(attribute.String("mac", mac.String())),
+	)
+
+	span.SetStatus(codes.Ok, filename)
+	span.End()
+
+	log = log.WithValues("filename", filename)
+	file, found := binary.Files[filename]
 	if !found {
-		s.Log.Info("requested file not found", "file", got)
+		log.Info("requested file not found")
 		http.NotFound(w, req)
 		return
 	}
 	b, err := w.Write(file)
 	if err != nil {
-		s.Log.Error(err, "error serving file")
+		log.Error(err, "error serving file")
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	s.Log.Info("file served", "bytes sent", b, "file size", len(file), "file", got)
+	log.Info("file served", "bytes sent", b, "file size", len(file))
```

Comment:
Is it normal/advisable to use variables like `bytes sent` as a semantic logging attribute name? I mean, should it be without spaces? e.g. `bytesSent`? will spaces make things harder to search for in a logging tool like elasticearch/loki?

Reply:
Great point! will update.